### ISSUE TITLE
Applet no longer connecting to yr.no

### ISSUE THREAD

**Issue by gbcox** [kotelnik/plasma-applet-weather-widget#89]
Within the last week, applet stopped updating.  I'm thinking it may be due to recent Qt / Plasma updates, but I suppose there is also the possibility that yr.no has changed something on their end.  In any event, I thought I'd let  you know so you could check it out.

Thanks!

**Comment by gbcox** [comment]
Figured it out.  yr.no has changed their website, so the code no longer works.  It needs to be be modified to match the new layout.  OWM works fine.

**Comment by sergiodurand** [comment]
@gbcox You can use the yr.no old website layout to copy the working URL.
https://retro.yr.no/

Before confirm the URL in widget window, remember to replace `retro` by `www`

(not sure for how long yr.no will keep the old URL working)

**Comment by sspanak** [comment]
This doesn't help. The widget does not accept URLs that begin with `retro.yr.no`.

Obviously, the code should be updated to parse the new website layout, but sadly, this would probably never happen. The project seems to be abandoned since 2017...

**Comment by sergiodurand** [comment]
> This doesn't help. The widget does not accept URLs that begin with `retro.yr.no`.


Before confirm the URL in widget window, **remember to replace retro by www**
Try again, it works that way.

**Comment by fkru** [comment]
> > This doesn't help. The widget does not accept URLs that begin with `retro.yr.no`.
> 
> Before confirm the URL in widget window, **remember to replace retro by www**
> Try again, it works that way.

Confirmed. However, the question remains whether the code will be adjusted to the new website.

**Comment by epassaro** [comment]
Probably adjusting [this line](https://github.com/kotelnik/plasma-applet-weather-widget/blob/02779f9cbf740a1a61776b904c5eb622788e6834/package/contents/ui/providers/YrNo.qml#L27) will make it work.

**Comment by sspanak** [comment]
> Probably adjusting [this line](https://github.com/kotelnik/plasma-applet-weather-widget/blob/02779f9cbf740a1a61776b904c5eb622788e6834/package/contents/ui/providers/YrNo.qml#L27) will make it work.

As @sergiodurand pointed out, there is no need to change anything in the code. Just taking away "retro" from the URL is fine. However, this would work only until `yr.no` shut down the retro site.

**Comment by fkru** [comment]
> > Probably adjusting [this line](https://github.com/kotelnik/plasma-applet-weather-widget/blob/02779f9cbf740a1a61776b904c5eb622788e6834/package/contents/ui/providers/YrNo.qml#L27) will make it work.
> 
> As @sergiodurand pointed out, there is no need to change anything in the code. Just taking away "retro" from the URL is fine. However, this would work only until `yr.no` shut down the retro site.

That's exactly the point, the retro.site will be removed sooner or later.

**Comment by epassaro** [comment]
Retro site should be online for a couple of years, otherwise does not have sense to make a legacy url. But this repo hasn't been updated in three years. Probably it's time to make a fork.

**Comment by fkru** [comment]
> 
> 
> Retro site should be online for a couple of years, otherwise does not have sense to make a legacy url. But this repo hasn't been updated in three years. Probably it's time to make a fork.

I inquired at yr.no on the availabilty of the retro site, and I have just got this answer:

"Meteorologisk institutt (Yr) 
Jan 20, 2021, 9:11 GMT+1 
We dont have any exact time.
Maybe april 2021.
Yr/ Elisabeth"

**Comment by JulienDlq** [comment]
I am interested too, I really like yr.no informations + meteogram!

**Comment by akontsevich** [comment]
Workaround:

1. Take URL from `https://retro.yr.no/place/...`
2. Copy paste to the widget
3. Replace `retro` prefix with `www`and voila - it works: `https://www.yr.no/place/...`

**Comment by akontsevich** [comment]
Hmm... stopped to work: was able to add URL once, all posterior URLs - widget do no add it.

**Comment by akontsevich** [comment]
> > This doesn't help. The widget does not accept URLs that begin with `retro.yr.no`.
> 
> Before confirm the URL in widget window, **remember to replace retro by www**
> Try again, it works that way.

Yep, that works!!! Thanks!

**Comment by JulienDlq** [comment]
Meteogram still not work with the workaround on my side. :cry:

**Comment by fkru** [comment]
> Workaround:
> 
>     1. Take URL from `https://retro.yr.no/place/...`
> 
>     2. Copy paste to the widget
> 
>     3. Replace `retro` prefix with `www`and voila - it works: `https://www.yr.no/place/...`

As I have already mentioned [above](https://github.com/kotelnik/plasma-applet-weather-widget/issues/89#issuecomment-763428383), the retro.yr site will be removed soon, maybe this month.

**Comment by akontsevich** [comment]
> As I have already mentioned [above](https://github.com/kotelnik/plasma-applet-weather-widget/issues/89#issuecomment-763428383), the retro.yr site will be removed soon, maybe this month.

Lucky people who'll keep old links :wink:

**Comment by fkru** [comment]
> > Workaround:
> > ```
> > 1. Take URL from `https://retro.yr.no/place/...`
> > 
> > 2. Copy paste to the widget
> > 
> > 3. Replace `retro` prefix with `www`and voila - it works: `https://www.yr.no/place/...`
> > ```
> 
> As I have already mentioned [above](https://github.com/kotelnik/plasma-applet-weather-widget/issues/89#issuecomment-763428383), the retro.yr site will be removed soon, maybe this month.

Just in case that the retro site is not available anymore and that the code of the widget is not adjusted accordingly: It also works with the new website if one replaces [...] in https://www.yr.no/[...]/Country/... by 'place'.

**Comment by alfaflo** [comment]
Now it seems the retro site has been shut down...
Anybody here who can modify "YrNo.qml" to be able to fetch the weather data from new web site? 🤔🙂

**Comment by gbcox** [comment]
I just switched over to using Open Weather Map.  Seems to work fine.

**Comment by alfaflo** [comment]
> I just switched over to using Open Weather Map. Seems to work fine.

Yes, OW works, but yr.no was very accurate in terms of weather forecast... I've loved it but I do not have the programming skills to "repair" yr.no data fetching... 🙁

**Comment by fkru** [comment]
> I just switched over to using Open Weather Map. Seems to work fine.

Probably the only solution, since the applet is not maintained anymore (tried to contact the maintainer some time ago without success). With yr.no the applet is dead.

**Comment by fkru** [comment]
The retro site is back! The applet is working as expected again.

**Comment by alfaflo** [comment]
> The retro site is back! The applet is working as expected again.

Very strange... couple of minutes ago it was "broken"  😁
Let's see how long it will work 🤔

**Comment by gbcox** [comment]
> > I just switched over to using Open Weather Map. Seems to work fine.
> 
> Probably the only solution, since the applet is not maintained anymore (tried to contact the maintainer some time ago without success). With yr.no the applet is dead.

The meteogram on the new yr site appears to be more like what is used on the OWM site - and says "under construction".  Not sure if that means they are working on making it look the same as the retro site or keep the "new" look.  If they keep the "new" look, I don't see any advantage of yr.no over OWM.  The retro meteogram on yr.no is very nice, but the new one isn't anything special.

**Comment by akontsevich** [comment]
> > I just switched over to using Open Weather Map. Seems to work fine.
> 
> Probably the only solution, since the applet is not maintained anymore 

The only solution is to fork this. :wink:

**Comment by arehtykitna** [comment]
> @gbcox You can use the yr.no old website layout to copy the working URL.
> https://retro.yr.no/
> 
> Before confirm the URL in widget window, remember to replace `retro` by `www`
> 
> (not sure for how long yr.no will keep the old URL working)

Until 4th October 2021, after that time the data set for the retro site will also be removed including the tables of weather data. Not sure how this will affect the Plasmoid but my current view is expect it to stop getting data as of that date or midnight on 5th October 2021.

**Comment by arehtykitna** [comment]
yr.no data has stopped working as of this morning so in effect the app is now specifically for OWM only. They scheduled this at yr.no some time ago and it has come to fruition for the app now, Wednesday onwards is showing N/A for the meteogram already. I will switch the the nearest OWM station too which isn't quite as accurate as the Norwegian Weather Service prediction but still better than BBC's.

**Comment by akontsevich** [comment]
Strange, still works for me!

**Comment by gbcox** [comment]
> Strange, still works for me!

Works for me also.  No issues.

**Comment by gbcox** [comment]
> > As I have already mentioned [above](https://github.com/kotelnik/plasma-applet-weather-widget/issues/89#issuecomment-763428383), the retro.yr site will be removed soon, maybe this month.
> 
> Lucky people who'll keep old links wink

As you previously mentioned, you need to keep the old links and replace retro.yr.no with www.yr.no.

**Comment by arehtykitna** [comment]
![Screenshot_20211004_221403](https://user-images.githubusercontent.com/7422283/135925777-a1a3ac79-7b6b-4ed5-b9ba-836b76142e1a.png)

**Comment by arehtykitna** [comment]
`[Containments][2][Applets][22][Configuration][General]
lastReloadedMsJson={"cache_5a5440a6c6026f3e61c6aee598cae8dc":1632806959604,"cache_63c4e139462a566e9c6718f387340bc7":1633381963936,"cache_d4e95df75abf69c7ea5dafc8415e1369":1633382019741}
placeIndex=1
places=[{"providerId":"owm","placeIdentifier":"7294618","placeAlias":"Betws yn Rhos, GB"},{"providerId":"yrno","placeIdentifier":"United_Kingdom/Wales/Betws_yn_Rhos","placeAlias":"Betws yn Rhos"}]`

**Comment by gbcox** [comment]
> `[Containments][2][Applets][22][Configuration][General] lastReloadedMsJson={"cache_5a5440a6c6026f3e61c6aee598cae8dc":1632806959604,"cache_63c4e139462a566e9c6718f387340bc7":1633381963936,"cache_d4e95df75abf69c7ea5dafc8415e1369":1633382019741} placeIndex=1 places=[{"providerId":"owm","placeIdentifier":"7294618","placeAlias":"Betws yn Rhos, GB"},{"providerId":"yrno","placeIdentifier":"United_Kingdom/Wales/Betws_yn_Rhos","placeAlias":"Betws yn Rhos"}]`

Are you asking a question?  You can't use retro.yr.no.  You have to use www.yr.no.

For example:  https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/

**Comment by arehtykitna** [comment]
> For example: https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/

No, I'm well aware of the need to do that. I'm just providing proof of that exact url not working any longer in the app. Look at the meteogram for Wednesday in the screenshot, the predictive table below it was generated with the last dataset available on the website but there is no detailed weather table for Wednesday onwards that the meteogram would be generated from.

The new site has them but the app cannot read them because it doesn't know where to load it from, they are using not only a new site but also a new server and backend.

**Comment by gbcox** [comment]
> > For example: https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/
> 
> No, I'm well aware of the need to do that. I'm just providing proof of that exact url not working any longer in the app. Look at the meteogram for Wednesday in the screenshot, the predictive table below it was generated with the last dataset available on the website but there is no detailed weather table for Wednesday onwards that the meteogram would be generated from.
> 
> The new site has them but the app cannot read them because it doesn't know where to load it from, they are using not only a new site but also a new server and backend.

I don't know what you're doing, but the URL that I provided:  https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/
works just fine.  
![Screenshot_20211004_172141](https://user-images.githubusercontent.com/4543019/135941682-21852475-a6f7-4e8f-a59c-5bb136ddc803.png)

**Comment by akontsevich** [comment]
Today yr.no stopped working for me.

**Comment by arehtykitna** [comment]
I let KDE render the meteogram rather than using the yr.no one in your screenshot because I find the yr.no meteogram too low contrast to see properly and jarring on the black UI background.

It's in the app settings to do so, however it cannot render off data it cannot find. But anyway, seeing as it has begun failing for other people it clearly cannot be attributed to 'what I'm doing' now can it.

**Comment by Sadi58** [comment]
It seems the good old http://www.yr.no website has finally gone for good now, and it's time to switch to somewhat less accurate OWP data - at least until someone forks this applet.

**Comment by fkru** [comment]
Yes, https://retro.yr.no/ does not exist anymore, but after some recent hiccup the weather widget just works fine. However, it is unclear to me how to add new links that provide the information delivered so far.

**Comment by gbcox** [comment]
> Yes, https://retro.yr.no/ does not exist anymore, but after some recent hiccup the weather widget just works fine. However, it is unclear to me how to add new links that provide the information delivered so far.

As mentioned previously, you have to rebuild the URL in this format:

https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/

If you saved your previous places, you should be fine.  All of my links are working.
![Screenshot_20211005_105359](https://user-images.githubusercontent.com/4543019/136077760-ada8ef66-8d36-43f0-96ad-08a5129363f3.png)

**Comment by akontsevich** [comment]
> > Yes, https://retro.yr.no/ does not exist anymore, but after some recent hiccup the weather widget just works fine. However, it is unclear to me how to add new links that provide the information delivered so far.
> 
> As mentioned previously, you have to rebuild the URL in this format:
> 
> https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/

Ha, yep, it works again, and previous links still work.

**Comment by Sadi58** [comment]
Weird. https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/ works while the new website displays it in this format: https://www.yr.no/en/forecast/daily-table/2-7299981/Great%20Britain/Wales/Conwy/Betws%20Yn%20Rhos
Luckily, I've figured out the correct URL format for my two locations now. ☺

**Comment by fkru** [comment]
> > Yes, https://retro.yr.no/ does not exist anymore, but after some recent hiccup the weather widget just works fine. However, it is unclear to me how to add new links that provide the information delivered so far.
> 
> As mentioned previously, you have to rebuild the URL in this format:
> 
> https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/
> 
> If you saved your previous places, you should be fine. All of my links are working. ![Screenshot_20211005_105359](https://user-images.githubusercontent.com/4543019/136077760-ada8ef66-8d36-43f0-96ad-08a5129363f3.png)

Does it still work for you? https://www.yr.no/place/United_Kingdom/Wales/Betws_yn_Rhos/ seems to be dead. Thx.

**Comment by arehtykitna** [comment]
Just over a week on and it's now stopped working permanently for the plasmoid for every location (with a message instead of a meteogram now) and for the last time I **had used the right url format** since it was posted about last December.

Since this is really abandonware anyway it's pointless even discussing this further or as someone else has done opening a new issue.

Stick with OWM or move to one of the standard KDE Weather plasmoid data source based Plasmoids.

I wish you all well in future endeavours!

**Comment by gbcox** [comment]
Current status as of October 13, 2021:  TLDR:  It works if you follow directions mentioned above.

![Screenshot_20211013_113926](https://user-images.githubusercontent.com/4543019/137194490-e645da6d-c9ba-4d83-b234-839fda3a09c5.png)

**Comment by fkru** [comment]
> Just over a week on and it's now stopped working permanently for the plasmoid for every location (with a message instead of a meteogram now) and for the last time I **had used the right url format** since it was posted about last December.
> 
> Since this is really abandonware anyway it's pointless even discussing this further or as someone else has done opening a new issue.
> 
> Stick with OWM or move to one of the standard KDE Weather plasmoid data source based Plasmoids.
> 
> I wish you all well in future endeavours!

It seems to be a seesaw situation. In fact, for me it's working again after a couple of hours. According to https://developer.yr.no/doc/guides/deprecating-old-widgets/, there is an alternative, which, however, does not work for me.

**Comment by gbcox** [comment]
> 
> It seems to be a seesaw situation. In fact, for me it's working again after a couple of hours. According to https://developer.yr.no/doc/guides/deprecating-old-widgets/, there is an alternative, which, however, does not work for me.

It's either their site (they are making ongoing changes), their internet provider, DNS or a mix and/or match of a multitude of factors.  I'm the OP and the fix listed in this thread has been working for me since December 2020.  Yes, occasionally I notice things aren't updating, but I just leave it alone and the next time I check it's back.  

I did read your post about the old widgets (thank you for the heads-up, BTW).  IMHO, the meteogram was the differentiator for the yr.no service.  If they are indeed going to deprecate it, I see no reason for me to continue to use their service.  

For those of you who might be interested in trying another KDE desktop widget, I would recommend:
https://github.com/Zren/plasma-applet-eventcalendar

It also will integrate your google calendar, etc. if you use that service.

**Comment by frankenfruity** [comment]
Throttling could explain why it periodically stops working for some. As far I as understand the TOS, this API shouldn't be used directly by clients. So this applet is kind of unsanctioned use of their service.
https://api.met.no/doc/TermsOfService

The API used by this applet is being phased out but for those of whom it's still working it's probably not great to use a short update interval. Their data only updates 4 times a day anyways for every where outside of Scandinavia.
https://hjelp.yr.no/hc/en-us/articles/360014549074-Updating-frequency-

**Comment by fkru** [comment]
I have installed version 2.1.4 of weather-widget-2 from https://store.kde.org/p/1683743, but it shows up only in the system tray, even though I used "add widget" to add it to the panel. Other widgets from KDE store are installed as expected. Anyone else seeing this? Any idea? Thx.
